sparse/sparse_array.py

```diff
@@ -20,7 +20,7 @@ class SparseArray(object):
 
     def __init__(self, shape):
         if not isinstance(shape, Iterable):
-            shape = (int(shape),)
+            shape = (shape,)
```

Comment:
Why this change?

Reply:
If we provided a simple unsupported class that wasn't `Iterable` as a shape, this would break but give an unhelpful error message about not being able to cast to `int`. This way, the error message is more informative.

I tend to bunch in small bugfixes along with big PRs in order to get them in quickly, I don't really know how good of a habit this is.

Reply:
It is common for numpy integers to sneak in here.  Sometimes these cause issues later on.  It would be good to ensure that all values of `shape` are eventually coerced to be normal Python integers.

Reply:
I anticipated this already. This is already done, a few lines below this one. :-)

It was being done here an extra time before the check. I just changed that so it happens after the check, and gives a meaningful error message.

https://github.com/pydata/sparse/blob/b9fc91cb68b37757f37881843b2e420a3c07f9be/sparse/sparse_array.py#L21-L29